pennylane/interfaces/execution.py

```diff
@@ -479,17 +482,50 @@ def cost_fn(params, x):
     else:
         batch_execute = set_shots(device, override_shots)(device.batch_execute)
 
-    expand_fn = _preprocess_expand_fn(expand_fn, device, max_expansion)
+    if transform_program is None:
+        expand_fn = _preprocess_expand_fn(expand_fn, device, max_expansion)

```

Comment:
```suggestion
    expand_fn = _preprocess_expand_fn(expand_fn, device, max_expansion)
```

Since this PR isn't sticking device preprocessing into the transform program, we can probably leave this line alone for right now. We could have a dedicated later PR where we move device preprocessing into the program at the `QNode` level.